File: 0989/0989.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2021/1/22 12:43
# File: 0989.py
# Desc: 
class Solution:
    def addToArrayForm(self, A: List[int], K: int) -> List[int]:
        return [int(i) for i in str(int("".join([str(j) for j in A])) + K)] if K else A

File: 1004/0989.java
/**
 * @Author: washing
 * @DateTime: 2021/02/19 17:11
 * @File: 1004.java
 * @Desc: 
 **/
class Solution {
    public int longestOnes(int[] A, int K) {
        int count = 0;
        int left = 0;

        for (int i : A) {
            if (i==0) count++;
            if (count>K) {
                if (A[left]==0) count--;
                left++;
            }
        }
        return A.length - left;
    }
}